modules/auxiliary/scanner/ssh/ssh_version.rb

```diff
@@ -4,96 +4,275 @@
 ##
 
 require 'recog'
+require 'net/ssh/transport/session'
 
 class MetasploitModule < Msf::Auxiliary
-  include Msf::Exploit::Remote::Tcp
   include Msf::Auxiliary::Scanner
   include Msf::Auxiliary::Report
 
-  # the default timeout (in seconds) to wait, in total, for both a successful
-  # connection to a given endpoint and for the initial protocol response
-  # from the supposed SSH endpoint to be returned
-  DEFAULT_TIMEOUT = 30
-
   def initialize
     super(
-      'Name'        => 'SSH Version Scanner',
-      'Description' => 'Detect SSH Version.',
-      'References'  =>
-        [
-          [ 'URL', 'https://en.wikipedia.org/wiki/SecureShell' ]
-        ],
-      'Author'      => [ 'Daniel van Eeden <metasploit[at]myname.nl>' ],
-      'License'     => MSF_LICENSE
+      'Name' => 'SSH Version Scanner',
+      'Description' => 'Detect SSH Version, and the server encryption',
+      'References' => [
+        ['URL', 'https://en.wikipedia.org/wiki/SecureShell'], # general info
+        ['URL', 'https://datatracker.ietf.org/doc/html/rfc8732#name-deprecated-algorithms'], # deprecation of kex gss-sha1 stuff
+        ['URL', 'https://datatracker.ietf.org/doc/html/draft-ietf-curdle-ssh-kex-sha2-20#page-16'], # diffie-hellman-group-exchange-sha1, diffie-hellman-group1-sha1, rsa1024-sha1
+        ['URL', 'https://datatracker.ietf.org/doc/html/rfc8758#name-iana-considerations'], # arc4 deprecation
+        ['URL', 'https://github.com/net-ssh/net-ssh?tab=readme-ov-file#supported-algorithms'], # a bunch of diff removed things from the ruby lib
+        ['CVE', '2008-5161'] # CBC modes
+      ],
+      'Author' => [
+        'Daniel van Eeden <metasploit[at]myname.nl>', # original author
+        'h00die' # algorithms enhancements
+      ],
+      'License' => MSF_LICENSE
     )
 
     register_options(
       [
         Opt::RPORT(22),
-        OptInt.new('TIMEOUT', [true, 'Timeout for the SSH probe', DEFAULT_TIMEOUT])
+        OptInt.new('TIMEOUT', [true, 'Timeout for the SSH probe', 30]),
+        OptBool.new('ENCRYPTION', [false, 'Check encryption for issues', true])
```

Comment:
```suggestion
        OptBool.new('EXTENDED_CHECKS', [true, 'Check for cryptographic issues', true])
```

The recog checks and host key checks are included in this, so calling it `ENCRYPTION` is just a little misleading for the user.

Reply:
i like this name a LOT better.